api/v1/storagecluster_types.go

```diff
@@ -249,6 +249,13 @@ type ManageCephBlockPools struct {
 // ManageCephNonResilientPools defines how to reconcile ceph non-resilient pools
 type ManageCephNonResilientPools struct {
 	Enable bool `json:"enable,omitempty"`
+	// UseExistingOsds is set when pools has to use existing OSDs
+	UseExistingOsds bool `json:"useExistingOsds,omitempty"`
+	// PoolNamePrefix is prefix string used in the replica-1 pools names
+	// +optional
+	PoolNamePrefix string `json:"poolNamePrefix,omitempty"`
+	// +optional
+	AutoCreateStorageclass bool `json:"autoCreateStorageclass,omitempty"`
 	// Count is the number of devices in this set
 	// +kubebuilder:validation:Minimum=1
 	Count int `json:"count,omitempty"`
```

Comment:
I believe,

```
	Count int `json:"count,omitempty"`
	// ResourceRequirements (requests/limits) for the devices
	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
	// VolumeClaimTemplates is a PVC template for the underlying storage devices
	VolumeClaimTemplate *corev1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
```

Is only needed to set when we are having new osds?

Reply:
Count and VolumeClaimTemplate only apply when using new OSDs. For this new feature, if `useExistingOsds: true`, these properties are just ignored.